homeassistant/components/matter/light.py

```diff
@@ -39,79 +71,328 @@ class MatterLight(MatterEntity, LightEntity):
 
     entity_description: MatterLightEntityDescription
 
+    def _supports_feature(
+        self, feature_map: int, feature: MatterColorControlFeatures
+    ) -> bool:
+        """Return if device supports given feature."""
+
+        return (feature_map & (1 << feature.value)) != 0
+
+    def _supports_color_mode(self, color_feature: MatterColorControlFeatures) -> bool:
+        """Return if device supports given color mode."""
+
+        if not self._supports_color():
+            return False
+
+        feature_map = self._device_type_instance.node.get_attribute(
+            self._device_type_instance.endpoint,
+            clusters.ColorControl,
+            clusters.ColorControl.Attributes.FeatureMap,
+        )
+
+        assert isinstance(feature_map.value, int)
+
+        return self._supports_feature(int(feature_map.value), color_feature)
+
+    def _supports_hs_color(self) -> bool:
+        """Return if device supports hs color."""
+
+        return self._supports_color_mode(MatterColorControlFeatures.HS)
+
+    def _supports_xy_color(self) -> bool:
+        """Return if device supports xy color."""
+
+        return self._supports_color_mode(MatterColorControlFeatures.XY)
+
+    def _supports_color_loop(self) -> bool:
+        """Return if device supports color loop."""
+
+        return self._supports_color_mode(MatterColorControlFeatures.COLOR_LOOP)
```

Comment:
this looks a bit weird to me as color loop is an effect/state the device is in and not a color mode. Maybe leave it out completely ? If someone is really attached to colorloops we can do a separate PR to introduce effect features ?

Reply:
Yeah removed the color loop part entirely since it's an optional thing that doesn't really need to be implemented and I don't think any manufacturer has implemented it in a first place